src/tests/JIT/Regression/JitBlue/Runtime_58373/Runtime_58373.cs

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
could you please tell me which complus are needed to repro it on x64 windows?

Reply:
I must have made a blunder somewhere, this test does not repro it for me either. It seems on x64 we normalize when we move the arg to the outgoing arg area/into arg register. So we need a little bit more. This exposes it on x64:
```csharp
using System;
using System.Runtime.CompilerServices;

public unsafe class Runtime_58373
{
    public static int Main()
    {
        short halfValue = HalfToInt16Bits(MakeHalf());
        int x = halfValue;
        short val2 = HalfToInt16Bits(*(Half*)&x);

        return halfValue == val2 ? 100 : -1;
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    static Half MakeHalf()
    {
        return (Half)(-1.0f);
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    static short HalfToInt16Bits(Half h)
    {
        return *(short*)&h;
    }
}  
```

This one exposes it on x86 without unsafe except for the reinterp:
```csharp
using System;
using System.Runtime.CompilerServices;

public unsafe class Runtime_58373
{
    public static int Main()
    {
        // Use up a lot of registers
        int a = GetVal();
        int b = GetVal();
        int c = GetVal();
        int d = GetVal();
        int e = GetVal();
        int f = GetVal();
        int g = GetVal();
        int h = GetVal();
        int i = GetVal();

        short val1 = HalfToInt16Bits(MakeHalf());
        Half half = MakeHalf();
        MakeHalf(); // This will spill lower 16 bits of 'half' to memory
        short val2 = HalfToInt16Bits(half); // This will pass 32 bits as arg with upper 16 bits undefined

        return val1 == val2 ? 100 + a + b + c + d + e + f + g + h + i : -1;
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    static int GetVal()
    {
        return 0;
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    static Half MakeHalf()
    {
        return default;
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    static short HalfToInt16Bits(Half h)
    {
        return *(short*)&h;
    }
}  
```
I will change the test to use these cases.